sdjournal/journal.go

```diff
@@ -86,6 +86,26 @@ func NewJournal() (*Journal, error) {
 	return j, nil
 }
 
+// NewJournalFromDir returns a new Journal instance pointing to a journal residing
+// in a given directory.  Typically, the directory will be appended with /<machine-id>
+// which can be generated from util.GetMachineID() from this same library.
+func NewJournalFromDir(path string) (*Journal, error) {
+	if len(path) == 0 {
+		return nil, fmt.Errorf("failed to open journal; provided path is nil:")
```

Comment:
Any reason to do this and not just let `sd_journal_open_directory` error out itself? Otherwise, we're required to update if they happen to change the API there and e.g. support empty string indicating current directory, or something like that.
Coming at it from the opposite direction, this could be an `AbsPath` check instead.